File: plugins/extbench/fixtures/pyvm-py/validate.py
# validate(payload) — same spec as the other validate fixtures.
# Payload: {"email": str, "age": number}
# Result : {"ok": bool, "email"?: str, "age"?: int, "error"?: str}
#
# Pure stdlib only — we're benchmarking the runtime, not the ecosystem.


def validate(p):
    if not isinstance(p, dict):
        return {"ok": False, "error": "input must be an object"}
    email = p.get("email")
    age = p.get("age")
    if not isinstance(email, str) or not email:
        return {"ok": False, "error": "email required"}
    if not isinstance(age, (int, float)) or age < 0 or age > 150:
        return {"ok": False, "error": "age out of range"}
    normalized = email.strip().lower()
    at = normalized.find("@")
    if at <= 0 or at == len(normalized) - 1:
        return {"ok": False, "error": "email shape"}
    domain = normalized[at + 1:]
    if "." not in domain:
        return {"ok": False, "error": "email domain"}
    return {"ok": True, "email": normalized, "age": age}

File: plugins/extbench/fixtures/v8go-js/validate.js
// Same logic as goja-js/index.js — only the runtime differs.
function validate(input) {
  if (input === null || typeof input !== "object") {
    return { ok: false, error: "input must be an object" };
  }
  var email = input.email;
  var age = input.age;
  if (typeof email !== "string" || email.length === 0) {
    return { ok: false, error: "email required" };
  }
  if (typeof age !== "number" || age < 0 || age > 150) {
    return { ok: false, error: "age out of range" };
  }
  var normalized = email.trim().toLowerCase();
  var at = normalized.indexOf("@");
  if (at <= 0 || at === normalized.length - 1) {
    return { ok: false, error: "email shape" };
  }
  var domain = normalized.substring(at + 1);
  if (domain.indexOf(".") < 0) {
    return { ok: false, error: "email domain" };
  }
  return { ok: true, email: normalized, age: age };
}
